components/sysutils/nut/nut-common.p5m

```diff
@@ -0,0 +1,45 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2016 Jim Klimov. All rights reserved.
+#
+
+set name=pkg.fmri \
+    value=pkg:/$(COMPONENT_FMRI)-common@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY) (Base accounts and dirs)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# Ensure the UID exists till the end, until after the main packages are removed
+# and in particular do not move databases and logs into lost+found on uninstall
+# Note the locked user password is NL (not NP) because it has to manage crontab
+group groupname=$(NUTGROUP) gid=77
+user ftpuser=false gcos-field="Network UPS Tools Reserved UID" group=$(NUTGROUP) \
+    password=NL username=$(NUTUSER) uid=77
+
+# These directories can hold some content generated or downloaded at runtime
+# and can become lost+found when you un/re-install the core packages (if these
+# pkg actions are delivered there instead)
+dir group=$(NUTGROUP) owner=$(NUTUSER) mode=0775 path=$(NUTETCDIR)
```

Comment:
Depends on the user's paranoia level ;)
Technically it does not have to e.g. edit config files (including its own).
But it should read sensitive ones such as nut password file. It also has to
write the killpower flag where shutdown routine can find it.

So it can be another user with same group and some posix bits on common
objects. In practice they are often conflated.

--
Typos courtesy of GMail on my Samsung Android
15.05.2016 7:07 пользователь "Alexander Pyhalov" <notifications@github.com>
написал:

> In components/sysutils/nut/nut-common.p5m
> <https://github.com/OpenIndiana/oi-userland/pull/2001#discussion_r63286267>
> :
>
> > +set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
> > +set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
> > +
> > +license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
> > +
> > +# Ensure the UID exists till the end, until after the main packages are removed
> > +# and in particular do not move databases and logs into lost+found on uninstall
> > +# Note the locked user password is NL (not NP) because it has to manage crontab
> > +group groupname=$(NUTGROUP) gid=77
> > +user ftpuser=false gcos-field="Network UPS Tools Reserved UID" group=$(NUTGROUP) \
> > +    password=NL username=$(NUTUSER) uid=77
> > +
> > +# These directories can hold some content generated or downloaded at runtime
> > +# and can become lost+found when you un/re-install the core packages (if these
> > +# pkg actions are delivered there instead)
> > +dir group=$(NUTGROUP) owner=$(NUTUSER) mode=0775 path=$(NUTETCDIR)
>
> shouldn't upsmon just run as nut?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/2001/files/8763ecacc8d248aa64f9eaebf64c307267ff7ab5#r63286267>
>